### PR TITLE
vnc: Decouple VNC server from NX graphic stack

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
since VNC server is very useful even without NX

- vnc: Decouple VNC server from NX graphic stack 
- grahpics: Move VNC server to drivers/video
- boards/sim: Add vnc config for test 

## Impact
vnc server can be used without enabling nx stack

## Testing
Pass new vnc config

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
Please ignore the false nxstyle warning:
```
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/drivers/video/vnc/vnc_keymap.c:123:1: error: Blank line precedes right brace at line
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/drivers/video/vnc/vnc_keymap.c:127:1: error: Blank line follows left brace
Error: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/drivers/video/vnc/vnc_keymap.c:130:1: error: Blank line precedes right brace at line
```

**Review by acassis — APPROVED** [review]

**Comment by acassis** [comment]
Very @xiaoxiang781216! Because we have VNC Server and VNC Client I suggest renaming boards/sim/sim/sim/configs/vnc/defconfig to boards/sim/sim/sim/configs/vncserver/defconfig

Did you test the VNC Client on NuttX?

**Comment by xiaoxiang781216** [comment]
> Very nice @xiaoxiang781216! Because we have VNC Server and VNC Client I suggest renaming boards/sim/sim/sim/configs/vnc/defconfig to boards/sim/sim/sim/configs/vncserver/defconfig
> 

Ok, I will update the config name

> Did you test the VNC Client on NuttX?

No, NuttX doesn't really implement VNC client now, just VNC server as a FB driver. VNC client is better to move to user space application. There are many 3rd party library can be ported there.

**Comment by xiaoxiang781216** [comment]
Done, @acassis please take a look.

File: include/nuttx/nx/nx.h [hunk 1]
```diff
@@ -258,6 +259,49 @@ extern "C"
...
+ *   Zero (OK) is returned on success.  Otherwise, a negated errno value is
+ *   returned to indicate the nature of the failure.
+ *
+ ****************************************************************************/
+
+/* int nx_vnc_fbinitialize(int display, NXHANDLE handle); */
+
+#if defined(CONFIG_NX_KBD) && defined(CONFIG_NX_XYINPUT)
+
+#define nx_vnc_fbinitialize(d,h) \
```
Comment (pkarashchenko) [inline include/nuttx/nx/nx.h:285]:
```suggestion
#  define nx_vnc_fbinitialize(d,h) \
```

File: include/nuttx/nx/nx.h [hunk 2]
```diff
@@ -258,6 +259,49 @@ extern "C"
...
+  vnc_fbinitialize((d), nx_kbdin, nx_mousein, (FAR void *)(h))
+
+#elif defined(CONFIG_NX_KBD)
+
+#define nx_vnc_fbinitialize(d,h) \
```
Comment (pkarashchenko) [inline include/nuttx/nx/nx.h:290]:
```suggestion
#  define nx_vnc_fbinitialize(d,h) \
```

File: include/nuttx/nx/nx.h [hunk 3]
```diff
@@ -258,6 +259,49 @@ extern "C"
...
+  vnc_fbinitialize((d), nx_kbdin, NULL, (FAR void *)(h))
+
+#elif defined(CONFIG_NX_XYINPUT)
+
+#define nx_vnc_fbinitialize(d,h) \
```
Comment (pkarashchenko) [inline include/nuttx/nx/nx.h:295]:
```suggestion
#  define nx_vnc_fbinitialize(d,h) \
```

File: include/nuttx/nx/nx.h [hunk 4]
```diff
@@ -258,6 +259,49 @@ extern "C"
...
+  vnc_fbinitialize((d), NULL, nx_mousein, (FAR void *)(h))
+
+#else
+
+#define nx_vnc_fbinitialize(d,h) \
```
Comment (pkarashchenko) [inline include/nuttx/nx/nx.h:300]:
```suggestion
#  define nx_vnc_fbinitialize(d,h) \
```

File: include/nuttx/nx/nx.h (hunk 1 above)
Comment (xiaoxiang781216) [reply include/nuttx/nx/nx.h]: Done.

File: include/nuttx/nx/nx.h (hunk 3 above)
Comment (xiaoxiang781216) [reply include/nuttx/nx/nx.h]: Done.

File: include/nuttx/nx/nx.h (hunk 4 above)
Comment (xiaoxiang781216) [reply include/nuttx/nx/nx.h]: Done.

File: include/nuttx/nx/nx.h (hunk 2 above)
Comment (xiaoxiang781216) [reply include/nuttx/nx/nx.h]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**